client/senddecorator/ratelimit_circuitbreaker.go

```diff
@@ -17,6 +17,11 @@ const (
 	defaultWaitAfterTooManyRequests = 6 * time.Minute
 )
 
+func init() {
+	// ONE DOES NOT SIMPLY RETRY ON HTTP 429.
+	autorest.StatusCodesForRetry = removeElementFromSlice(autorest.StatusCodesForRetry, http.StatusTooManyRequests)
+}
```

Comment:
I know this is personal, but I still it's _magical_ to use `init()`. What do you say about moving `autorest.StatusCodesForRetry = removeElementFromSlice(autorest.StatusCodesForRetry, http.StatusTooManyRequests)` to the first line of the `RateLimitCircuitBreaker` function, right above the return?

Reply:
My reasoning for `init()` is that it's guaranteed to run once before actual business logic whenever this package is imported. It's a global variable in library and changing that repeatedly on function call feels semantically wrong to me. `init()` makes more sense since that knob should be turned once and be left at that.

Reply:
okok